openstack/baremetal/v1/nodes/requests.go

```diff
@@ -371,3 +387,52 @@ func GetSupportedBootDevices(client *gophercloud.ServiceClient, id string) (r Su
 	})
 	return
 }
+
+// A cleaning step has required keys ‘interface’ and ‘step’, and optional key ‘args’. If specified,
+// the value for ‘args’ is a keyword variable argument dictionary that is passed to the cleaning step
+// method.
+type CleanStep struct {
+	Interface string            `json:"interface,required"`
```

Comment:
Sorry, I just noticed the `required` piece here.

AFAIK, Go doesn't have a `required` JSON tag. Gophercloud supports a very basic one: https://github.com/gophercloud/gophercloud/blob/master/params.go#L64

```go
Interface string `json:"interface" required:"true"`
```

Reply:
Apologies, my mistake.  I also did that on node updates, I opened a new PR with tests here: https://github.com/gophercloud/gophercloud/pull/1482

I'll update this one as well.